src/Adapter/FirehoseAdapter.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * This file is part of graze/queue.
+ *
+ * Copyright (c) 2015 Nature Delivered Ltd. <https://www.graze.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license https://github.com/graze/queue/blob/master/LICENSE MIT
+ *
+ * @link    https://github.com/graze/queue
+ */
+
+namespace Graze\Queue\Adapter;
+
+use Aws\Firehose\FirehoseClient;
+use Graze\Queue\Adapter\Exception\MethodNotSupportedException;
+use Graze\Queue\Adapter\Exception\FailedEnqueueException;
+use Graze\Queue\Message\MessageFactoryInterface;
+use Graze\Queue\Message\MessageInterface;
+
+/**
+ * Amazon AWS Kinesis Firehose Adapter.
+ *
+ * This method only supports the enqueue method to send messages to a Kinesiss
+ * Firehose stream
+ *
+ */
+final class FirehoseAdapter implements AdapterInterface
+{
+    const BATCHSIZE_SEND    = 100;
+
+    /** @var FirehoseClient */
+    protected $client;
+
+    /** @var array */
+    protected $options;
+
+    /** @var string */
+    protected $deliveryStreamName;
+
+    /**
+     * @param FirehoseClient $client
+     * @param string    $deliveryStreamName
+     * @param array     $options
+     */
+    public function __construct(FirehoseClient $client, $deliveryStreamName, array $options = [])
+    {
+        $this->client = $client;
+        $this->deliveryStreamName = $deliveryStreamName;
+        $this->options = $options;
+    }
+
+    /**
+     * @param MessageInterface[] $messages
+     *
+     * @throws MethodNotSupportedException
+     */
+    public function acknowledge(array $messages)
+    {
+        throw new MethodNotSupportedException(
+            'acknowledge',
+            $this,
+            $messages
+        );
+    }
+
+    /**
+     * @param MessageFactoryInterface $factory
+     * @param int                     $limit
+     *
+     * @throws MethodNotSupportedException
+     */
+    public function dequeue(MessageFactoryInterface $factory, $limit)
+    {
+        throw new MethodNotSupportedException(
+            'dequeue',
+            $this,
+            []
+        );
+    }
+
+    /**
+     * @param MessageInterface[] $messages
+     *
+     * @throws FailedEnqueueException
+     */
+    public function enqueue(array $messages)
+    {
+        $failed = [];
+        $batches = array_chunk($this->createEnqueueEntries($messages), self::BATCHSIZE_SEND);
```

Comment:
we probably don't do this currently. But we should allow this field to be overloaded.

an option in the constructor options, or argument or getter/setter.

Reply:
I've add this as an option and added an options accessor method.